packages/interaction-tracking/src/InteractionTracking.js

```diff
@@ -0,0 +1,341 @@
+    interaction.__count = 1;
+
+    let caughtError;
+    let didCatch = false;
+    let returnValue;
+    const subscriber = ((subscriberRef: any): SubscriberRef).current;
+
+    if (subscriber !== null) {
+      try {
+        subscriber.onInteractionTracked(interaction);
+        subscriber.onWorkStarted(interactions, threadID);
+      } catch (error) {
+        didCatch = true;
+        caughtError = caughtError || error;
+      }
+    }
+
+    try {
+      returnValue = callback();
+    } catch (error) {
+      didCatch = true;
+      caughtError = caughtError || error;
+    }
+
+    ((interactionsRef: any): InteractionsRef).current = prevInteractions;
+
+    if (subscriber !== null) {
+      try {
+        subscriber.onWorkStopped(interactions, threadID);
+      } catch (error) {
+        didCatch = true;
+        caughtError = caughtError || error;
+      }
+    }
+
+    interaction.__count = ((interaction.__count: any): number) - 1;
+
+    // If no async work was scheduled for this interaction,
+    // Notify subscribers that it's completed.
+    if (subscriber !== null && ((interaction.__count: any): number) === 0) {
+      try {
+        subscriber.onInteractionScheduledWorkCompleted(interaction);
+      } catch (error) {
+        didCatch = true;
+        caughtError = caughtError || error;
+      }
+    }
+
+    if (didCatch) {
+      throw caughtError;
+    } else {
+      return returnValue;
+    }
+  } else {
+    try {
+      return callback();
+    } finally {
+      ((interactionsRef: any): InteractionsRef).current = prevInteractions;
+    }
+  }
+}
+
+export function unsubscribe(subscriber: Subscriber): void {
+  if (enableInteractionTracking && enableInteractionTrackingObserver) {
+    ((subscriberRef: any): SubscriberRef).current = null;
+  }
+}
+
+export function wrap(
+  callback: Function,
+  threadID: number = DEFAULT_THREAD_ID,
+): Function {
+  if (!enableInteractionTracking) {
+    return callback;
+  }
+
+  const wrappedInteractions = ((interactionsRef: any): InteractionsRef).current;
+
+  if (enableInteractionTrackingObserver) {
+    const subscriber = ((subscriberRef: any): SubscriberRef).current;
+    if (subscriber !== null) {
+      subscriber.onWorkScheduled(wrappedInteractions, threadID);
+    }
+
+    // Update the pending async work count for the current interactions.
+    // Update after calling subscribers in case of error.
+    wrappedInteractions.forEach(interaction => {
+      interaction.__count = ((interaction.__count: any): number) + 1;
+    });
+  }
+
+  const wrapped = (...args) => {
+    const prevInteractions = ((interactionsRef: any): InteractionsRef).current;
+    ((interactionsRef: any): InteractionsRef).current = wrappedInteractions;
+
+    if (enableInteractionTrackingObserver) {
+      const subscriber = ((subscriberRef: any): SubscriberRef).current;
+
+      try {
+        let caughtError;
+        let didCatch = false;
```

Comment:
Same thing here. You can just use nested try/finally.

Reply:
Ditto ^